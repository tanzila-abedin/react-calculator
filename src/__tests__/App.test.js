import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Calculator from '../components/App';

test(' Snapshot Calculator app and nested componets', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('App renders Wrapper component', async () => {
  const app = render(<Calculator />);
  const wrapper = await app.findAllByTestId('wrapper');
  expect(wrapper).toBeDefined();
});
