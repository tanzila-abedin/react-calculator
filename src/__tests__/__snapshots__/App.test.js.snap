// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Snapshot Calculator app and nested componets 1`] = `
<div
  className="container"
>
  <h4
    className="text-center top-head pb-0"
  >
    <strong>
      Calculator App
    </strong>
  </h4>
  <h5
    className="text-center sub-head pt-0"
  >
    Built by Tanzila Abedin
  </h5>
  <div
    className="d-flex justify-content-center p-2"
    data-testid="wrapper"
  >
    <div
      className="wrapper"
    >
      <div
        className="display"
      >
        0
      </div>
      <div
        className="buttonPanel"
      >
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          +-
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          /
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className=""
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="equals"
          data-testid="test-btn"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
